### PR TITLE
Support alter statement ALGORITHM and LOCK options parse

### PR DESCRIPTION
**Pull request by zihaoAK47**
Refer #25862.

Changes proposed in this pull request:
  - Support alter statement ALGORITHM and LOCK options

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ✓] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ✓] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ✓] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ✓] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by zihaoAK47** [comment]
@TeslaCN Mentor please review.

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/enums/AlgorithmOption.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+@RequiredArgsConstructor
+@Getter
+public enum AlgorithmOption {
+    
+    DEFAULT("DEFAULT"),
+    INSTANT("INSTANT"),
+    INPLACE("INPLACE"),
+    COPY("COPY");
+    
+    private final String algorithmOption;
```
Comment (TeslaCN) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/enums/AlgorithmOption.java:35]: Is this field necessary?

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/enums/LockTableOption.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+@RequiredArgsConstructor
+@Getter
+public enum LockTableOption {
+    
+    DEFAULT("DEFAULT"),
+    NONE("NONE"),
+    SHARED("SHARED"),
+    EXCLUSIVE("EXCLUSIVE");
+    
+    private final String lockTableOption;
```
Comment (TeslaCN) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/enums/LockTableOption.java:35]: Is this field necessary?

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLAlterTableStatement.java [hunk 3]
```diff
@@ -17,11 +17,21 @@
...
 /**
  * MySQL alter table statement.
  */
+@Getter
+@Setter
 public final class MySQLAlterTableStatement extends AlterTableStatement implements MySQLStatement {
+    
+    private AlgorithmTypeSegment algorithmSegment;
+    
+    private LockTableSegment lockTableSegment;
```
Comment (TeslaCN) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLAlterTableStatement.java:36]: Please remove the `@Getter` annotation, and add `Optional` getter method for them.

**Comment by TeslaCN** [comment]
@zihaoAK47 Sorry for my unclear description.

I mean the field `String algorithmOption` could be removed, not the enum. Just like
```java
public enum AlgorithmOption {

     DEFAULT,
     INSTANT,
     INPLACE,
     COPY
}
```

**Comment by zihaoAK47** [comment]
@TeslaCN Mentor, the check failed. I don't know how this error occurred. I have completed the modifications and the local build has passed

**Comment by TeslaCN** [comment]
> @TeslaCN Mentor, the check failed. I don't know how this error occurred. I have completed the modifications and the local build has passed

Thank you. The failure was not caused by this PR. I'm going to merge this soon.

**Review by TeslaCN — APPROVED** [review]

**Review by TeslaCN — APPROVED** [review]

**Pull request merged**